axum-macros/src/typed_path.rs

```diff
@@ -18,52 +18,82 @@ pub(crate) fn expand(item_struct: ItemStruct) -> syn::Result<TokenStream> {
         ));
     }
 
-    let Attrs { path } = parse_attrs(attrs)?;
+    let Attrs { path, rejection } = parse_attrs(attrs)?;
 
     match fields {
         syn::Fields::Named(_) => {
             let segments = parse_path(&path)?;
-            Ok(expand_named_fields(ident, path, &segments))
+            Ok(expand_named_fields(ident, path, &segments, rejection))
         }
         syn::Fields::Unnamed(fields) => {
             let segments = parse_path(&path)?;
-            expand_unnamed_fields(fields, ident, path, &segments)
+            expand_unnamed_fields(fields, ident, path, &segments, rejection)
         }
-        syn::Fields::Unit => expand_unit_fields(ident, path),
+        syn::Fields::Unit => expand_unit_fields(ident, path, rejection),
     }
 }
 
+mod kw {
+    syn::custom_keyword!(rejection);
+}
+
 struct Attrs {
     path: LitStr,
+    rejection: Option<syn::Path>,
+}
+
+impl Parse for Attrs {
+    fn parse(input: syn::parse::ParseStream) -> syn::Result<Self> {
+        let path = input.parse()?;
+
+        let _ = input.parse::<Token![,]>();
+
+        let lh = input.lookahead1();
+        let rejection = if lh.peek(kw::rejection) {
+            input.parse::<kw::rejection>()?;
+            let content;
+            syn::parenthesized!(content in input);
+            Some(content.parse()?)
+        } else if lh.is_empty() {
+            None
+        } else {
+            return Err(lh.error());
+        };
```

Comment:
Okay so this should be fully correct 😅

Otherwise `#[typed_path("foo" rejection(Rejection))]` would work I think.

```suggestion
        let rejection = if input.is_empty() {
            None
        } else {
            let _: Token![,] = input.parse()?;
            let _: kw::rejection = input.parse()?;

            let content;
            syn::parenthesized!(content in input);
            Some(content.parse()?)
        };
```